README.md

```diff
@@ -1,29 +1,343 @@
+   ```
+
+3. Select the version that you want to install. While the SDK is its alpha stage, we recommend that you install *an exact* version of the SDK instead of indicating a range. This prevents you from automatically upgrading to a newer alpha version and possibly installing breaking changes on your code base. 
+
+   ![in xcode select Exact from dropdown when selecting the version of the SDK to install](docs/img/xcode_spm_install_version.jpg)
+
+4. Lastly, choose the SDK products that you want to install. You can start by selecting `Tracking` for now and adding others later if you need them. 
+
+![](docs/img/xcode_spm_select_sdks.jpeg)
+
+## Initialize the SDK
+
+Before you can use the Customer.io SDK, you need to initialize it. Any calls that you make to the SDK before you initialize it are ignored. 
+
+There are two ways to initialize the SDK. The method you use depends on how you want to use the SDK:
+
+1. Singleton, shared instance (the quick and easy way):
+
+When you use the shared instance, you don't need to manage your own instances of Customer.io SDK classes. To get started, initialize the SDK in the `AppDelegate` `application(_ application: didFinishLaunchingWithOptions)` function: 
+
+```swift
+import CioTracking
+
+class AppDelegate: NSObject, UIApplicationDelegate {
+    func application(
+        _ application: UIApplication,
+        didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]? = nil
+    ) -> Bool {
+        CustomerIO.initialize(siteId: "YOUR SITE ID", apiKey: "YOUR API KEY")
+
+        // You can optionally provide a Region to set the Region for your Workspace:
+        CustomerIO.initialize(siteId: "YOUR SITE ID", apiKey: "YOUR API KEY", region: Region.EU)
+
+        return true
+    }
+}
+```
+
+Then, when you want to use any of the SDK features, you use the shared instance of the class:
+
+```swift
+MessagingPush.shared.application(...)
+```
+
+2. Create your own instances (better for projects using automated tests):
+
+We recommend that you create your own instances of SDK classes if your project has automated tests. We designed our SDK with first-class support for dependency injection and mocking, which makes it easier to write automated tests. See [testing](#Testing) for more information.
+
+> **Note**: Code samples in this readme use the singleton, shared instance method to call the SDK. However, all samples will also work with your own instances of SDK classes.
+
+```swift
+import CioTracking
+
+let customerIO = CustomerIO(siteId: "XXX", apiKey: "YYY")
+
+// You can optionally provide a Region to set the Region for your Workspace:
+let customerIO = CustomerIO(siteId: "XXX", apiKey: "YYY", region: Region.EU)
+```
+
+Then, when you want to use any of the SDK features, you use the shared instance of the class:
+
+```swift
+let messagingPush = MessagingPush(customerIO: customerIO)
+
+messagingPush.application(...)
+```
+
+# Tracking
+
+## Identify a customer
+
```

Comment:
Should we note right here that you need to identify someone to truly take advantage of the SDK? Like, it's running, but since we don't support anonymous devices/users, you really do need to identify someone to attribute a token to them, send a push, etc?